Campus-iOS/App.swift

```diff
@@ -40,6 +41,9 @@ struct CampusApp: App {
                     }
                     .navigationViewStyle(.stack)
                 }
+                .onReceive(model.$isLoginSheetPresented, perform: { isLoginSheetPresented in
```

Comment:
Why exactly is this needed? Any bugs otherwise?

Reply:
Because of this new warning (see image). This is since Xcode 14/SwiftUI 4 and does not allow anymore to update the view by changing properties of the ViewModel. I don't know if this means we have to have tons of new `@State` properties in every view, but as far as I found out this is currently the only solution to dismiss this warning. Also see https://www.youtube.com/watch?v=3a7tuhVpoTQ.

<img width="500" alt="Screenshot 2022-10-17 at 12 28 20" src="https://user-images.githubusercontent.com/59373377/196160690-0fb38102-cb44-439a-a113-bdf8264fdb47.png">

Reply:
Yeah this should probably be done via the `DispatchQueue`

Reply:
Something like `DispatchQueue.main.async`

Reply:
And then pass the respective closure to this function

Reply:
Ok, now it works, but why does this need to be on the main thread? ☺️

Reply:
For one issue it workend and for this line of code it didn't. We need to discuss this on thursday 😊

Reply:
I still get the warning despite wrapping every call into a `Dispatch.main.async{}`-call 😢.
We should discuss this on thursday. I opened another branch [fix/publish-warning](https://github.com/TUM-Dev/Campus-iOS/tree/fix/publish-warning).

<img width="500" alt="Screenshot 2022-10-17 at 16 09 34" src="https://user-images.githubusercontent.com/59373377/196199306-2cb62347-aff9-475d-bc21-d710d6005a03.png">